### PR TITLE
Add preference to limit the number of recent workspaces listed

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Added a `workspace preference` to limit the number of recent workspace listed for end users.

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/workspace/src/browser/quick-open-workspace.ts [hunk 1]
```diff
@@ -80,7 +82,7 @@ export class QuickOpenWorkspace implements QuickOpenModel {
         this.items = [];
         this.workspaceService.recentWorkspaces().then(workspaceRoots => {
             if (workspaceRoots) {
```
Comment (elaihau) [inline packages/workspace/src/browser/quick-open-workspace.ts]: this is not part of your change, but `workspaceRoots` is an array and wouldn't be null/undefined as per the return type of `workspaceService.recentWorkspaces`. therefore there's no need to do the if-check.

**Comment by akosyakov** [comment]
I wonder:
- do we really need a preference for it - have you ever seen it in other IDEs or configured it?
- should not we just make sure that we never store more than 10 recent workspaces on the backend?

**Comment by vince-fugnitto** [comment]
@akosyakov I guess it never really hurts to have additional customization options in an IDE to tailor to different use cases and types of users but if you don't see any added value we don't need to necessarily add a preference for `recent workspace limit`. I have seen customization options for recent history in **Notepad ++** and **Atom** for instance.

Example: Atom
selection_001


One improvement in the future that I'll likely add is to not only list recent workspaces, but have the ability to `individually delete` some from the history along with a `clear all`.

**Comment by marechal-p** [comment]
Yes, for people using a Theia application as a standard desktop IDE, it may help to have an exhaustive workspace history.

Now so far we don't have a collapsible menu bar item for it (like `Recent Workspaces >`) where we would hard limit the list to 10. But in the quickopen menu we can display as much as we want, with the search it wouldn't be so bad: if you look for something it will be there.

So I don't have any strong opinion about having a preference for this. I would not limit it with the way it currently is.

**Comment by vince-fugnitto** [comment]
maybe the focus should instead be on implementing the ability to `clear recent workspace entries` along with a `clear all`? This way the user will have the ability to manage his own history and will align with `vscode`.

**Pull request closed without merging**